core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -5792,6 +5798,13 @@ object ZIO extends ZIOCompanionPlatformSpecific {
       }
   }
 
+  final class LogAnnotations[V](val key: () => LogAnnotation[V], val value: () => V) {
```

Comment:
also here, I'd rename this to `LogAnnotation` as its just one key-value pair

Reply:
ok, as we already have that class name in the same package, maybe `LogAnnotationEntry` ( from Map->Entries )?